### DISCUSSION TITLE
Set force-cache when using cookie

### DISCUSSION THREAD

**Discussion by J4v4Scr1pt** [vercel/next.js#57846] [Help]
### Summary

The issue is that during the migration period we need to get a cookie containing the user session for AspNetCore.
And we need to provide this cookie on every fetch to our api.

I have created a custom GET function to reduce DRY.
But because I use cookies() NextJs opt-out of caching the request. Which is not desirble in this case.

To fix this I just force the cache and this works. 

Except the obvious one that all calls will be cached now are there any drawbacks doing this?

```
export async function GET(endpoint: string) {
	const cookieStore = cookies();
	const aspNetCoreCookies = cookieStore
		.getAll()
		.map((c) => `${c.name}=${c.value}`)
		.join(';');
	try {
		const res = await fetch(`${process.env.PATH_INTERNAL}${endpoint}`, {
			method: 'GET',
			cache: 'force-cache',
			headers: {
				'Content-Type': 'application/json',
				Cookie: aspNetCoreCookies || '',
			},
		});
		return await res.json();
	} catch (err) {
		return null;
	}
}
```

**Comment by FacundoPellicer** [comment]
Hello @J4v4Scr1pt , what did you end up doing?

I am in a similar situation, but the `force-cache` option when using cookies() does not work in production, although it does work locally.

**Reply by J4v4Scr1pt** [reply]
Hi man!

I still use force-cache, did not know it did not work in Prod. Because the [Docs ](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#caching-data)states it's _"// 'force-cache' is the default, and can be omitted"_ 
But maybe use React cache instead then?

**Reply by FacundoPellicer** [reply]
Thanks for the response!

If I am not mistaken, React cache is for duplicate requests. It is like a "dedupe", it is "Per-request lifecycle".

Yeah, maybe I have some bug, and thats why in my app the "force-cache" option is not working in prod, when used after a cookie() call.

Could you check, if it’s not too much trouble, if your app in production is being cached with the use of cookies and “force-cache”?

thanks anyways!

**Reply by J4v4Scr1pt** [reply]
Np!

That could be the case with react cache, I have not deepdived into that to much yet. maybe unstable_cache could work then? (no deepdive on that either tough 😅)

I can for sure check! But not until Friday.
How did you check it? Im hosting on Amazon, but have DataDog for logging.

**Reply by FacundoPellicer** [reply]
@J4v4Scr1pt Hello, I just wanted to let you know that after some investigation, I decided not to cache these fetches because it’s not recommended to cache user-specific data. I’m not entirely sure why, but one reason could be security.

I still don’t know why it doesn’t cache the fetches in production, because according to the docs, it should.

Thanks for your help!

**Reply by J4v4Scr1pt** [reply]
@FacundoPellicer great it was solved in at least one way 😊.
